runtime/lua/vim/lsp.lua

```diff
@@ -401,7 +402,6 @@ local function on_client_exit(code, signal, client_id)
         })
       end
 
-      local namespace = vim.lsp.diagnostic.get_namespace(client_id)
       vim.diagnostic.reset(namespace, bufnr)
```

Comment:
why doesn't this work? surely the purpose (at least) of creating a per-client namespace is to be able to clear or reset the related diagnostics without a bunch of extra code.

Reply:
Actually, I think you are correct. I reread the docs for vim.diagnostic.reset and it does say when the buffer number is omitted, diagnostics are removed for all buffers. I will test if that works as that would be much cleaner.

Reply:
You were correct vim.diagnostic.reset does have the ability to clear all diagnostics on the namespace without needing to manually iterate over all the buffers. I cleaned up the code with that fix.